recipe-client-addon/lib/FilterExpressions.jsm

```diff
@@ -41,3 +43,23 @@ this.FilterExpressions = {
     return jexl.eval(onelineExpr, context);
   },
 };
+
+/**
+ * Return an array of the given object's own keys (specifically, its enumerable
+ * properties).
+ * @param {Object} obj
+ * @return {Array[String]}
+ */
+function keys(obj) {
+  return Object.keys(obj);
+}
+
+/**
+ * Find all the values that are present in both lists.
+ * @param {Array} listA
+ * @param {Array} listB
+ * @return {Array}
+ */
+function operatorIntersect(listA, listB) {
+  return listA.filter(item => listB.includes(item));
```

Comment:
Yep, `this` is a funny thing. [It is set to the object that the function is called on, _not_ the object that it is defined on.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this#As_an_object_method). So from `filter`'s perspective, it's getting a standalone function and calling it directly; it doesn't know about `list2`, and so `this` isn't set unless we bind the function.